### DISCUSSION TITLE
selfServiceError - type of error object

### DISCUSSION THREAD

**Discussion by woylie** [ory/kratos#2136] [Q&A] [answered]
According to the [http api doc](https://www.ory.sh/kratos/docs/reference/api#operation/getSelfServiceError), the `selfServiceError` has a field called `error`, which is described as a generic object. While trying out Kratos, I got this map under the `error` field:

```elixir
%{
  "code" => code,
  "message" => message,
  "reason" => reason,
  "status" => status
}
```

This map would match the description of `genericError`. Hence the question: Is the value of the `error` field always a `genericError`? If so, I'll submit a PR to update the OpenAPI spec. Also, the `error` field is marked as optional, but it would seem odd to only return an ID on that route without an error. Should the field be marked as required?

**Comment by vinckr** [comment, marked as answer]
Hello @woylie 

>Is the value of the error field always a genericError?

Maybe I misunderstand your question, but that endpoint allows you to implement the error page in any way you want, so you can use the genericError as template for error messages you want to show. But I get the feeling that was not your question?

>the error field is marked as optional

Maybe I am looking at it wrong, but it seems the error and message are both marked as required in the API DOC? 
If this is different in the actual map, this is potentially a bug; could you open an issue to clarify  it?

**Reply by woylie** [reply]
Hello @vinckr,

I'm referring to the documentation here:

<img width="439" alt="imagen" src="https://user-images.githubusercontent.com/13847569/150322959-fec28601-721c-457d-9d46-cf3aac477a1c.png">

The `error` field is just listed as an object without any field descriptions, and it is not marked as required. The question is whether this `error` field should be marked as required and be described as conforming to the `genericError` type, just as the `error` field of the `jsonError` type is:

<img width="428" alt="imagen" src="https://user-images.githubusercontent.com/13847569/150323437-df24c51d-2c4f-4378-8269-a46402b993f1.png">

**Reply by vinckr** [reply]
I see, thanks for clarifying. 
Is this a potential bug in the OpenAPI spec @zepatrik ?

**Reply by vinckr** [reply]
You took it into your own hands and opened a PR that solves it, a true hero 🙇 🎉 
https://github.com/ory/kratos/pull/2375

**Reply by woylie** [reply]
@vinckr I didn't get an answer and didn't change anything related to this. I assume this is the line that would have to be changed: https://github.com/ory/kratos/blob/master/selfservice/errorx/error.go#L23. Currently, the type is a `json.RawMessage`, but I still don't know whether this should indeed be a GenericError.

**Reply by vinckr** [reply]
Oh my bad, then your PR was unrelated, still thanks for looking into it. 
edit: I am also not sure if it should be GenericError.

**Reply by aeneasr** [reply]
Hey there - you’re richt, in most (all?) cases the referenced struct should be correct!

**Reply by woylie** [reply]
> Hey there - you’re richt, in most (all?) cases the referenced struct should be correct!

Sorry, are you saying that `selfServiceError.error` should be documented as a `genericError` instead of just an `object`, or are you saying that the documentation is correct as it is?

**Reply by aeneasr** [reply]
> selfServiceError.error should be documented as a genericError instead of just an object

that is what i intended to say
